src/js/models/layerstructure.js

```diff
@@ -826,6 +826,19 @@ define(function (require, exports, module) {
     }));
 
     /**
+     * Determine whether a vector mask can be created on a layer 
+     *
+     * @param {Layer} layer
+     * @return {boolean}
+     */
+    Object.defineProperty(LayerStructure.prototype, "canHaveVectorMask", objUtil.cachedLookupSpec(function (layer) {
```

Comment:
I thought that we only supported this for a given layer selection? Can this just be a property and not a lookup? I.e., shouldn't this always operate on the given layer selection? For example: if no layers are selected, then the answer is no. If many layers are selected, then the answer hinges on the first selected layer, right?